src/mpid/ch4/src/ch4_types.h

```diff
@@ -307,6 +311,7 @@ typedef struct MPIDI_CH4_Global_t {
     int my_sigusr1_count;
 #endif
     OPA_int_t progress_count;
```

Comment:
Why do we have two counters?  One for `progress_count` and one for `progress_attempt_counter`?

Reply:
The `progress_count` is incremented whenever an `MPIR_Request` is freed. The purpose of this counter is to check if the state of the progress engine has changed since it began poking progress for a `MPID_Progress_wait`.

I think changing the name of the `progress_attempt_counter` to `vci_progress_attempt_counter` would help with readability.

Reply:
`progress_count` sounds like a boolean to me, telling the progress engine if any progress was made or not.  I think that's the variable that needs to be renamed.